temporalio/client.py

```diff
@@ -4259,27 +4354,33 @@ async def result(
             ),
         )
 
-        # Continue polling as long as we have either an empty response, or an *rpc* timeout
+        # Continue polling as long as we have no outcome
         while True:
-            try:
-                res = (
-                    await self._client.workflow_service.poll_workflow_execution_update(
-                        req,
-                        retry=True,
-                        metadata=rpc_metadata,
-                        timeout=rpc_timeout,
-                    )
-                )
-                if res.HasField("outcome"):
-                    return await _update_outcome_to_result(
-                        res.outcome,
-                        self.id,
-                        self._client.data_converter,
-                        self._result_type,
-                    )
-            except RPCError as err:
-                if err.status != RPCStatusCode.DEADLINE_EXCEEDED:
-                    raise
+            res = await self._client.workflow_service.poll_workflow_execution_update(
+                req,
+                retry=True,
+                metadata=rpc_metadata,
+                timeout=rpc_timeout,
+            )
+            if res.HasField("outcome"):
+                self._known_outcome = res.outcome
+                return
+
+
+class WorkflowUpdateWaitStage(IntEnum):
+    """Stage to wait for workflow update to reach before returning from
```

Comment:
Another comment that mostly restates the name of the enum.

Reply:
This is commonly the case with all forced-documentation docs, which includes most of the Python docs. We can't rely on self-documenting names. Similar thing exists in Go. If there is a general different posture we want to take with the Python API docs, I wonder if we can do that generally. I am totally supportive of such a general posture change, just not necessarily for this one enum in this one PR.